src/otx/core/utils/tile_merge.py

```diff
@@ -0,0 +1,215 @@
+    img: np.ndarray,
+    img_info: ImageInfo,
+    merged_item: DatasetItem,
+) -> InstanceSegPredEntity:
+    """Create merged instance segmentation prediction.
+
+    Args:
+        img (np.ndarray): Original untiled image.
+        img_info (ImageInfo): Image information.
+        merged_item (DatasetItem): Merged dataset item.
+
+    Returns:
+        InstanceSegPredEntity: Merged inst-seg prediction.
+    """
+    device = img_info.device
+    pred_bboxes, pred_labels, pred_scores, pred_mask_instances = [], [], [], []
+    pred_masks_by_label = {}
+    for anno in merged_item.annotations:
+        if isinstance(anno, Bbox):
+            pred_bboxes.append(anno.points)
+            pred_labels.append(anno.label)
+            pred_scores.append(float(anno.attributes["score"]))
+        if isinstance(anno, Mask):
+            # NOTE: Datumaro tile merge does not give mask instances.
+            # It merges mask instances to one mask.
+            pred_masks_by_label[anno.label] = anno.image
+
+    if len(pred_bboxes) == 0:
+        pred_bboxes = torch.empty((0, 4))
+    else:
+        for pred_box, pred_label in zip(pred_bboxes, pred_labels):
+            x1, y1, x2, y2 = (int(value) for value in pred_box)
+            pred_label_mask = pred_masks_by_label[pred_label]
+            # TODO (Eugene): Performance issue here if there are too many mask instances.
+            # Ideally to optimize memory and speed we should convert it to RLE or save the crop pred_mask[y1:y2, x1:x2]
+            # https://github.com/openvinotoolkit/datumaro/pull/1194
+            bitmask = np.zeros_like(pred_label_mask)
+            bitmask[y1:y2, x1:x2] = pred_label_mask[y1:y2, x1:x2]
+            pred_mask_instances.append(bitmask)
+
+    return InstanceSegPredEntity(
+        image=tv_tensors.Image(img),
+        img_info=img_info,
+        score=torch.tensor(pred_scores, device=device),
+        bboxes=tv_tensors.BoundingBoxes(pred_bboxes, format="XYXY", canvas_size=img.shape[:2], device=device),
+        labels=torch.tensor(pred_labels, device=device),
+        masks=tv_tensors.Mask(pred_mask_instances, dtype=torch.bool, device=device),
+        polygons=[],
+    )
+
+
+def merge_detection_tiles(
+    tile_preds: list[DetBatchPredEntity],
+) -> DetPredEntity:
+    """Merge detection tiles into one single detection prediction.
+
+    Args:
+        tile_preds (list[DetBatchPredEntity]): List of tile predictions.
+
+    Returns:
+        DetPredEntity: Merged detection prediction.
+    """
+    dataset_items = []
+    anno_id = 0
+
+    for tile_pred in tile_preds:
+        annotations = []
+        for batch_bboxes, batch_labels, batch_scores in zip(tile_pred.bboxes, tile_pred.labels, tile_pred.scores):
+            for bbox, label, score in zip(batch_bboxes.data, batch_labels, batch_scores):
+                _bbox = bbox.detach().cpu().numpy()
+                _label = label.detach().cpu().numpy()
+                _score = score.detach().cpu().numpy()
```

Comment:
Can we optimize it by having `torch.Tensor` operation-based un-tiling methods in the future (not in this PR)?

Reply:
yea, it would be ideal to have a tile merge function that is compatible with PyTorch tensors. Currently, tile merge relies on datumaro, so there are two potential approaches: either create custom tile merge utilities in OTX or request support for torch.tensor in datumaro tile merge. I will make a note for future reference to address this.